go/teams/chain.go

```diff
@@ -665,7 +662,65 @@ func (t *TeamSigChainPlayer) addInnerLink(prevState *TeamSigChainState, link SCC
 
 		return res, nil
 	case "team.leave":
-		return res, fmt.Errorf("todo implement parsing of: %s", payload.Body.Type) // TODO CORE-5305
+		if prevState == nil {
+			return res, fmt.Errorf("link type '%s' unexpected at seqno:%v", payload.Body.Type, prevState.LastSeqno+1)
+		}
+		if len(team.ID) == 0 {
+			return res, errors.New("missing team id")
+		}
+		if team.Name != nil {
+			return res, errors.New("unexpected name")
+		}
+		if team.Members != nil {
+			return res, errors.New("unexpected members")
+		}
+		if team.Parent != nil {
+			return res, errors.New("unexpected parent")
+		}
+		if team.Subteam != nil {
+			return res, errors.New("unexpected subteam")
+		}
```

Comment:
I wonder if we could factor some of this field checking into a single function, so the different types don't need to repeat it? Not for this PR though.